File: src/main/java/xyz/capybara/clamav/commands/VersionCommands.java
package xyz.capybara.clamav.commands;

import xyz.capybara.clamav.exceptions.InvalidResponseException;

import java.util.Arrays;
import java.util.Collection;
import java.util.stream.Collectors;

public class VersionCommands extends Command<Collection<String>> {

    public static final String COMMAND = "VERSIONCOMMANDS";
    private static final String COMMANDS_START_TAG = "| COMMANDS:";

    @Override
    public String getCommandString() {
        return COMMAND;
    }

    @Override
    protected CommandFormat getFormat() {
        return CommandFormat.NEW_LINE;
    }

    @Override
    protected Collection<String> parseResponse(String responseString) {
        int commandsStartPos = responseString.indexOf(COMMANDS_START_TAG);

        if (commandsStartPos == -1) {
            throw new InvalidResponseException(responseString);
        }

        return Arrays.stream(responseString.substring(commandsStartPos + COMMANDS_START_TAG.length()).split(" "))
                .filter(s -> !s.isEmpty())
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/xyz/capybara/clamav/commands/VersionCommands.kt
package xyz.capybara.clamav.commands

import xyz.capybara.clamav.InvalidResponseException

internal object VersionCommands : Command<Collection<String>>() {
    private const val COMMANDS_START_TAG = "| COMMANDS:"

    override val commandString
        get() = "VERSIONCOMMANDS"

    override val format
        get() = CommandFormat.NEW_LINE

    override fun parseResponse(responseString: String): Collection<String> {
        return when (val commandsStartPos = responseString.indexOf(COMMANDS_START_TAG)) {
            -1 -> throw InvalidResponseException(responseString)
            else -> responseString.substring(commandsStartPos + COMMANDS_START_TAG.length).split(" ".toRegex())
        }
    }
}
